controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -72,23 +74,77 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	clientSet, err := kubeclientset.NewForConfig(cachedCluster.RestConfig)
 	if err != nil {
 		reqLogger.Error(err, "cannot create ClientSet for the ToolchainCluster")
+		if err := r.updateStatus(ctx, toolchainCluster, clusterOfflineCondition(err.Error())); err != nil {
+			reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
+		}
 		return reconcile.Result{}, err
 	}
-	healthChecker := &HealthChecker{
-		localClusterClient:     r.Client,
-		remoteClusterClient:    cachedCluster.Client,
-		remoteClusterClientset: clientSet,
-		logger:                 reqLogger,
-	}
+
+	// execute healthcheck
+	healthCheckResult := r.getClusterHealthCondition(ctx, clientSet)
+
 	// update the status of the individual cluster.
-	if err := healthChecker.updateIndividualClusterStatus(ctx, toolchainCluster); err != nil {
+	if err := r.updateStatus(ctx, toolchainCluster, healthCheckResult); err != nil {
 		reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
 		return reconcile.Result{}, err
 	}
-
 	return reconcile.Result{RequeueAfter: r.RequeAfter}, nil
 }
 
+func (r *Reconciler) updateStatus(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster, currentConditions ...toolchainv1alpha1.Condition) error {
+	toolchainCluster.Status.Conditions = condition.AddOrUpdateStatusConditionsWithLastUpdatedTimestamp(toolchainCluster.Status.Conditions, currentConditions...)
+	if err := r.Client.Status().Update(ctx, toolchainCluster); err != nil {
+		return fmt.Errorf("failed to update the status of cluster - %s "+err.Error(), toolchainCluster.Name)
```

Comment:
let's use proper formatting
```suggestion
		return fmt.Errorf("failed to update the status of cluster - %s: %w", toolchainCluster.Name, err)
```